### PR TITLE
Set the context path and the readinessProbe Automatically

### PR DESCRIPTION
**Pull request by bordenit**
The helm chart currently does not allow for a context path to be used, as the readinessProbe will never become ready as it will not point to the correct context path. This change is intended to set the ATL_TOMCAT_CONTEXTPATH and the readinessProbe path dynamically based on the context path  provided in the values.yaml.

### CODE REVIEW

**Comment by bordenit** [comment]
Relates to issue #85

File: src/main/charts/jira/templates/statefulset.yaml [hunk 1]
```diff
@@ -73,6 +73,8 @@ spec:
                 configMapKeyRef:
                   key: reserved_code_cache
                   name: {{ include "jira.fullname" . }}-jvm-config
+            - name: ATL_TOMCAT_CONTEXTPATH
```
Comment (pbruski) [inline src/main/charts/jira/templates/statefulset.yaml:76]: Should we fix Confluence chart too?

**Review by pbruski — APPROVED** [review]

File: src/main/charts/jira/templates/statefulset.yaml (hunk 1 above)
Comment (nanux) [reply src/main/charts/jira/templates/statefulset.yaml:76]: I have created https://github.com/atlassian-labs/data-center-helm-charts/pull/90 to provide support for Confluence. The Bitbucket is still missing it as it provides access via Spring boot properties but I believe the `healthinessProbe` will still need to be changed.

**Comment by nanux** [comment]
@bordenit, thank you for your contribution. I believe if you run `mvn clean install` the build will fail as the helm output for Jira won't be as expected. I have created https://github.com/atlassian-labs/data-center-helm-charts/pull/90 that is passing and should give you an idea of what changes are required to make it passing. I have also created a test specifically for the context path value and it could be used by Jira as well. I can do that in my PR after we merge this.

Could you please fix the current tests before we are ready to merge?
Thank you

File: src/main/charts/confluence/templates/_helpers.tpl [hunk 2]
```diff
@@ -149,6 +149,10 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 -Datlassian.logging.cloud.enabled={{.Values.fluentd.enabled}}
 {{- end }}
 
+{{- define "confluence.sysprop.enable.synchrony.by.default" -}}
```
Comment (nanux) [inline src/main/charts/confluence/templates/_helpers.tpl:152]: Why is this as part of the PR?

**Review by nanux — CHANGES_REQUESTED** [review]

**Review by kennymacleod — CHANGES_REQUESTED** [review]
Remove irrelevant changes related to synchrony

**Comment by bordenit** [comment]
@nanux  Thanks. I addressed the comments and it looks like the tests passed.

**Comment by nanux** [comment]
> @nanux Thanks. I addressed the comments and it looks like the tests passed.

Thank you, could you please add it to the existing test? It is `@ParametrizedTest` so we can leverage set of products:

```
    @ParameterizedTest
    @EnumSource(value = Product.class, names = {"confluence", "jira"})
    void confluence_context_path(Product product) throws Exception {
        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
                "confluence.service.contextPath", "/" + product.name()));

        resources.getStatefulSet(product.getHelmReleaseName())
                .getContainer()
                .getEnv()
                .assertHasValue("ATL_TOMCAT_CONTEXTPATH", "/" + product.name());


        assertEquals(resources.getStatefulSet(
                product.getHelmReleaseName()).getContainer().get("readinessProbe").get("httpGet").get("path").asText(),
                "/" + product.name() + "/status");
    }
```

**Comment by bordenit** [comment]
@nanux I hope that worked, but if not, feel free to take it from here.

**Review by nanux — APPROVED** [review]
Thank you for the changes to the test and your contribution.

**Pull request merged**